opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java

```diff
@@ -444,6 +447,11 @@ public void validateRule(final String rule) throws FilterParseException {
         isRuleMatching(rule);
     }
 
+    @Override
+    public Map<Integer, String> getNodeLocations() {
```

Comment:
There is no guarantee that nodeLocations are loaded before you call this method. We can't depend upon another method to be called before.